rfcs/0038-sudt-info/0038-sudt info.md

```diff
@@ -0,0 +1,185 @@
+
+For this type of SUDT, you can construct an info_cell while minting SUDT, whose typescript's code_hash is info_cell typescript, its args is SUDT's type_script hash. Info_cell will check whether the transaction meets the conditions.
+
+````
+// Issue new SUDT/ create a new SUDT_Info cell
+Inputs:
+  <... one of the input cells must have owner lock script as lock>
+Outputs:
+	SUDT_Cell:
+		Data:
+			amount: uint128
+        Type:
+        	code_hash: simple_UDT type script
+     		args: owner lock script hash (...)
+     	Lock:
+     		<user defined>
+	SUDT_Info_Cell:
+  		Data:
+  			SUDT Info
+  		Type:
+  			code_hash: SUDT_Info type script
+  			args: SUDT type script hash
+  		Lock:
+  			SUDT creator defined
+````
+
+The following rules should be met in a SUDT Info Cell (typescript):
+
+- **Rule 1:** validate the format of info cell data.
+- **Rule 2:** In this transaction, at least one SUDT cell must exist in the output cell, and the hash of its typescript matches the args of Info_cell.
+- **Rule 3:** In this transaction, at least one cell must exist in the input cell, and its lockscript is the owner lockscript of SUDT.
+- **Rule 4:** the lockscript of info_cell should be set to deadlock by default, in some cases, developers can choose other lockscript.
+- **Rule 5:** If there are multiple Info_cells, choose the info_cell with the smallest block height at the time of generation. If it is at the same block height, choose the info_cell with the smallest transaction index. The same transaction cannot generate two info_cells.
+
+### 3.2. For Script-drive SUDT
+
+However, there is another type of SUDT on CKB, which is not issued by a lockscript controlled by a private key but follows a specific script logic. Such as NexisDAO's dCKB and Tai, and the previous Info_cell design is no longer suitable for this situation.
+
+Since anyone can mint SUDT while following the script logic, if we continue to use the previous logic, anyone can generate the corresponding info_cell. So we need to add new logic on the basis of the previous design.
+
+Since it is impossible to distinguish whether an owner lockscript is script-driven from the outside, it should be checked first according to the script-driven logic.
+
+````
+// Create a new SUDT_Info cell
+Inputs:
+	 ...
+Outputs:
+	SUDT_Info_Cell:
+		Data:
+			SUDT Info
+		Type:
+			code_hash: SUDT_Info type script
+			args: type_id
+		Lock:
+			SUDT creator defined
+````
+
+The following rules should be met in a SUDT Info Cell (typescript):
+
+- **Rule 1:** validate the format of info cell data.
+- **Rule 2:** The args conform to the rules of type id.
+- **Rule 3:** The lockscript of info_cell should be set to deadlock by default, in some cases, developers can choose other lockscript.
+- **Rule 4:** After constructing this Info cell, use the hash of info_cell's type_script as the first 32 bytes of the args of the lockscript of SUDT owner. 
```

Comment:
For this rule, there is no way to add sudt info for existing script-drive SUDT with this protocol, right?

Reply:
Yes, info is bound to the token.

Reply:
It might be a problem. Since token from NexisDAO and Force Bridge can not use this protocol then.

Reply:
I consulted Nexisdao, they have designed this, and udt info typescript can't actually sense whether a udt is controlled by a private key or not, it should first judge according to the script-driven mode, and then judge according to the single-owner mode , so for the info cell of udt originating from forcebridge, it can be generated according to the mode of single owner, as long as it is the earliest one.